src/client.js

```diff
@@ -304,11 +304,19 @@ Client.prototype.updateSession = function(token, ip, request) {
 		friendlyAgent += ` on ${agent.os.name} ${agent.os.version}`;
 	}
 
-	client.config.sessions[token] = _.assign({
+	client.config.sessions[token] = _.assign(client.config.sessions[token], {
 		lastUse: Date.now(),
 		ip: ip,
 		agent: friendlyAgent,
-	}, client.config.sessions[token]);
+	});
+
+	client.manager.updateUser(client.name, {
+		sessions: client.config.sessions
+	}, (err) => {
+		if (err) {
+			log.error("Failed to update sessions for", client.name, err);
```

Comment:
Isn't that going to log multiple lines like we've seen in other places? If so, you might have to use something like this:

```js
log.error(`Failed to update sessions for ${color.green(client.name)} (${err})`);
```